js/web/lib/wasm/jsep/webgpu/ops/matmul.ts

```diff
@@ -160,10 +161,12 @@ export const createNaiveMatmulProgramInfo = (
   }
   `;
   };
+  // Input broadcasting impacts the shader code, and should be handled in cache key
+  const inputBroadcastingDims = `${broadcastADims};${broadcastBDims}`;
```

Comment:
Since aShape and bShape are already uploaded as uniform, can you reuse those information to get broadcasted info to avoid adding new shader key info?

Reply:
Done.